src/browser/ui/dom/components/ReactDOMInput.js

```diff
@@ -115,6 +138,24 @@ var ReactDOMInput = ReactClass.createClass({
   _handleChange: function(event) {
     var returnValue;
     var onChange = LinkedValueUtils.getOnChange(this.props);
+
+    // IE 10+ fire input events when setting/unsetting a placeholder
+    // we guard against it by checking if the next and
+    // last values are both empty and bailing out of the change
+    // https://github.com/facebook/react/issues/3484
+    if (isIEInputEvent(event)) {
+      var controlledValue = LinkedValueUtils.getValue(this.props);
+      var lastValue = controlledValue != null ?
+        '' + controlledValue :
+        this._uncontrolledValue;
+
+      this._uncontrolledValue = event.target.value;
+
+      if ( event.target.value === '' && lastValue === '') {
+        return returnValue;
```

Comment:
yeah, honestly i returned `returnValue` just to appease eslint's consistent-return rule, originally it was just a lone `return`, which would be less prone to accidental refactoring breakage. Alternatively I could just do wrap the whole thing in a big if/else or return undefined perhaps? Not sure how to best handle the lint rule :P